tensorflow/core/kernels/mkl_pooling_ops_common.h

```diff
@@ -548,12 +548,23 @@ class MklPoolingOpBase : public OpKernel {
     if (pool_params->data_format == TensorFormat::FORMAT_NCHW) {
       output_tf_shape = MklDnnDimsToTFShape(output_dims_mkl_order);
     } else {
-      memory::dims output_dims_NHWC_order;
-      output_dims_NHWC_order = {pool_params->tensor_in_batch,
-                                static_cast<int>(pool_params->out_height),
-                                static_cast<int>(pool_params->out_width),
-                                pool_params->out_depth};
-      output_tf_shape = MklDnnDimsToTFShape(output_dims_NHWC_order);
+      // determine Pooling2D (NHWC) or Pooling3D (NDHWC)
+      if (this->ksize_.size() == 4) {
+        memory::dims output_dims_NHWC_order;
```

Comment:
Would you mind refactoring this and the `MklDnnDimsToTFShape` line out from `if`?
```c++
memory::dims output_dims_order;
if (...) {
  ...
} else {
  ...
} 
output_tf_shape = MklDnnDimsToTFShape(output_dims_order);
```

Reply:
I will do the refactoring, thanks!

Reply:
Refactoring done, thanks again!